### PR TITLE
webapp/functionapp: support cors

### LINKED ISSUE #4565: Azure cli functionapp and webapp has no support for CORS control
Like the title says, the cli provides me no mechanism to administer the cors whitelist for the functionapp.

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #4565 
Expose `az functionapp cors add/remve/show` and `as webapp cors add/remove/show`. This is to be consistent with portal, which expose "cors" as a top level tab

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -1371,6 +1371,29 @@ def clear_traffic_routing(cmd, resource_group_name, name):
     set_traffic_routing(cmd, resource_group_name, name, [])
 
 
+def add_cors(cmd, resource_group_name, name, allowed_origins):
```
Comment (panchagnula) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Need to support all these commands for slots as well.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py [hunk 2]
```diff
@@ -565,6 +565,37 @@ def test_traffic_routing(self, resource_group):
...
+class AppServiceCors(ScenarioTest):
+    @ResourceGroupPreparer()
+    def test_webapp_cors(self, resource_group):
+        self.kwargs.update({
+            'plan': self.create_random_name(prefix='slot-traffic-plan', length=24),
+            'web': self.create_random_name(prefix='slot-traffic-web', length=24),
+        })
+        self.cmd('appservice plan create -g {rg} -n {plan} --sku S1')
+        self.cmd('webapp create -g {rg} -n {web} --plan {plan}')
+        self.cmd('webapp cors add -g {rg} -n {web} --allowed-origins https://msdn.com https://msn.com')
```
Comment (panchagnula) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py]: once we add support for slots can you add another test here as well for slots? Thanks!

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Want to confirm it is supported with slot. I thought about that, but in portal there is no UI gesture for slot. Anyway, if yes, i will make the change

Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Yes it is supported on slots for both: on the portal if you go to Deployment Slots you will see CORS in the menu blade. & supported on Functions as well
image

Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Ah, got it I only tried out the webapp with slots, and the tab of "API->CORS" doesn't have the drop down for me to tweak the slot specific origins. Funcapp's ui makes more sense.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 3]
```diff
@@ -388,6 +388,26 @@
...
+
+helps['webapp cors add'] = """
+    type: command
+    short-summary: Add allowed origins
+"""
+
+helps['webapp cors remove'] = """
+    type: command
+    short-summary: Remove allowed origins
+"""
```
[the remark this answers is not in the document]
Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: I added an example. Note, users have no way to pass an empty array, as this flag require values to be provided. I can introduce `*` meaning to clear all, but I would suggest we do that based on user voices, unless it is common that you might have 10+ origins added that you like to clear them all

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Done for webapp. Note `deployment slot` as a feature, is not yet exposed for functionapp, so even though my change has already made cors capable for functionapp slot, but for consistency we should not make it visible till we enable `az functionapp deployment slot`. This way users would have full e2e under `az functionapp`. Sounds reasonable?
Also, when slot support for functionapp will default to be enabled?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py (hunk 2 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py]: done

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 4]
```diff
@@ -388,6 +388,26 @@
...
 """
 
+helps['webapp cors'] = """
+    type: group
+    short-summary: Manage Cross-Origin Resource Sharing (CORS)
+"""
+
+helps['webapp cors add'] = """
+    type: command
+    short-summary: Add allowed origins
```
[the remark this answers is not in the document]
Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: done

**Pull request merged**